src/transxchange/read.rs

```diff
@@ -168,15 +186,105 @@ fn load_companies(transxchange: &Element) -> Result<CollectionWithId<Company>> {
     Ok(companies)
 }
 
+fn load_commercial_physical_modes(
+    transxchange: &Element,
+) -> Result<(CommercialMode, PhysicalMode)> {
+    let raw_mode = transxchange
+        .try_only_child("Services")?
+        .try_only_child("Service")?
+        .try_only_child("Mode")?
+        .text();
+    let mode = MODES
+        .get(raw_mode.as_str())
+        .ok_or_else(|| {
+            format_err!(
+                "Failed to identify the commercial/physical mode from '{}'",
+                raw_mode
+            )
+        })?
+        .to_string();
+    let commercial_mode = CommercialMode {
+        id: mode.clone(),
+        name: mode.clone(),
+    };
+    let physical_mode = PhysicalMode {
+        id: mode.clone(),
+        name: mode.clone(),
+        ..Default::default()
+    };
+    Ok((commercial_mode, physical_mode))
+}
+
+fn load_line(transxchange: &Element, network_id: &str, commercial_mode_id: &str) -> Result<Line> {
+    let service = transxchange
+        .try_only_child("Services")?
+        .try_only_child("Service")?;
+    let id = service.try_only_child("ServiceCode")?.text();
```

Comment:
Concatenate ServiceCode and Lineid (several lines possible with the same ServiceCode)

Reply:
Done in 4990a7ad2e6561f105f021feb354944130648383.